app/src/main/res/layout/fragment_tab_tray.xml

```diff
@@ -0,0 +1,98 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:mozac="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <TextView
+        android:id="@+id/tab_tray_empty_view"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        android:gravity="center"
+        android:text="@string/no_open_tabs_description"
+        android:textColor="?secondaryText"
+        android:textSize="16sp"
+        android:visibility="gone"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent" />
+
+    <mozilla.components.concept.tabstray.TabsTray
+        android:id="@+id/tabsTray"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        app:layout_constraintBottom_toTopOf="@+id/saveToCollection"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        mozac:tabsTrayItemTextColor="?tabTrayItemText"
+        mozac:tabsTraySelectedItemTextColor="?tabTrayItemSelectedText"
+        mozac:tabsTrayItemBackgroundColor="?tabTrayItemBackground"
+        mozac:tabsTraySelectedItemBackgroundColor="?tabTrayItemSelectedBackground" />
+
+    <LinearLayout
+        android:id="@+id/saveToCollection"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:paddingStart="16dp"
+        android:paddingEnd="16dp"
+        app:layout_constraintTop_toBottomOf="@+id/tabsTray"
+        app:layout_constraintBottom_toTopOf="@id/bottomBarShadow">
+        <include
+            layout="@layout/save_to_collection_button"
+            android:id="@+id/saveToCollectionButton"/>
+    </LinearLayout>
+
+    <View
+        android:id="@+id/bottomBarShadow"
+        android:layout_width="0dp"
+        android:layout_height="1dp"
+        android:background="@color/bottom_bar_shadow"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintBottom_toTopOf="@id/tab_tray_controls" />
+
+    <LinearLayout
```

Comment:
Why are we using a linear layout for these buttons? Probably better to do a nested constraint in terms of performance?